openssl/src/symm.rs

```diff
@@ -381,6 +381,49 @@ impl Crypter {
         }
     }
 
+    /// Returns the cipher in use
+    pub fn cipher(&self) -> Cipher {
+        unsafe { Cipher(ffi::EVP_CIPHER_CTX_cipher(self.ctx)) }
+    }
+
+    /// Resets state, allowing a new encryption or decryption to take place.
+    ///
+    /// This is more efficient than creating a new `Crypter`.
+    pub fn reset(&mut self, mode: Mode, key: &[u8], iv: Option<&[u8]>) -> Result<(), ErrorStack> {
+        let mode = match mode {
```

Comment:
We should have the constructor call into this to avoid code duplication if possible.